### DISCUSSION TITLE
What is the best connection setup to use Nav2 and SLAM toolbox with create 3

### DISCUSSION THREAD

**Discussion by omerdurmus61** [iRobotEducation/create3_docs#553] [Questions]
### How are you connecting to your Create 3?

Wi-Fi and USB Ethernet (ROS 2)

### Computer(s) Model(s) and Operating System(s)

Intel Nuc and Desktop

### Which version of ROS 2 is installed on your computer?

Humble

### Which firmware version is installed on your robot?

H.2.4

### Which RMW is your robot running?

FastRTPS

### Does your robot have an assigned namespace? If so, please share.

No

### Is the robot connected to a network? If so, what is the network type?

office

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

None

### What is the Adapter Board's USB/BLE Toggle currently switched to?

USB

### Describe your question.

I have an Intel NUC mini PC connected to my Create 3 via an Ethernet cable. I adjusted the wired connection settings on the mini PC and used an Ethernet to USB-C adapter for the connection. I can access Create 3 topics over both Wi-Fi and the Ethernet cable on the mini PC. Additionally, I have a desktop PC that can access Create 3 topics over Wi-Fi.

However, when working with the Nav2 stack on both the mini PC and desktop, I encounter some errors likely due to delays and my Nav2 stack doesn't work properly. The errors are as follows:

Do you have any advice on resolving these errors?

```
[INFO] [1707460066.543602087] [amcl]: Message Filter dropping message: frame 'laser_frame' at time 1707460065.407 for reason 'the timestamp on the message is earlier than all the data in the transform cache

Message Filter dropping message: frame 'base_footprint' at time 1707466052.645 for reason 'discarding message because the queue is full'
```

**Comment by brianabouchard** [comment]
Hello! I've experienced similar errors in the past. One of the biggest improvements I saw was when I setup an NTP server. I was doing it on a Raspberry Pi, but the setup you have with your NUC mini PC sounds the same so it should be possible to use the same instructions to do so: https://iroboteducation.github.io/create3_docs/setup/compute-ntp/

The other option that can greatly improve performance, especially on an office/corporate network, would be implementing the Fast-DDS discovery server. iRobot has instructions on how to do that here: https://iroboteducation.github.io/create3_docs/setup/discovery-server/
